monix-tail/shared/src/main/scala/monix/tail/Iterant.scala

```diff
@@ -1888,6 +1898,25 @@ object Iterant extends IterantInstances {
   def pure[F[_], A](a: A): Iterant[F, A] =
     now[F, A](a)
 
+  def bracket[F[_], A, B](acquire: F[A])(
+    use: A => Iterant[F, B],
+    release: (A, BracketResult) => F[Unit]
+  )(
+    implicit F: Sync[F]
+  ): Iterant[F, B] = {
+    Iterant.liftF(acquire).flatMap { a =>
+      Iterant.suspend(F.map(F.attempt(F.delay(use(a))))(_.fold(
+        ex => Iterant.raiseError[F, B](ex),
+        iterant => iterant
```

Comment:
This weird machinery is required for lifting errors `use` might throw directly into `Iterant`. As a result, it will be reported as `Halt(Some(ex))` inside `Iterant[F, ?]` context . Alternatively, `release` could be called directly after `use` fails, and the error would be reported inside `F` itself. I went for the former.

See the test
> Bracket releases resource if `use` throws